xarray/core/extension_array.py

```diff
@@ -142,3 +142,6 @@ def __array__(
             return np.asarray(self.array, dtype=dtype, copy=copy)
         else:
             return np.asarray(self.array, dtype=dtype)
+
+    def __getattr__(self, attr: str) -> Any:
+        return getattr(self.array, attr)
```

Comment:
```suggestion
        return getattr(super().__getattribute__("array"), attr)
```

Reply:
I'd usually use `super().__getattr__("array")` or `object.__getattr__(self, "array")` in this case to break the recursion.

According to the data model pages, [`__getattr__`](https://docs.python.org/3/reference/datamodel.html#object.__getattr__) is only called if `__getattribute__` or `__get__` return an `AttributeError`, while `__getattribute__` is always called (with the exception of special cases like language syntax or builtin functions).

Reply:
Both options gave something like this, which makes sense to me but it's possible I'm missing something.  Would love to understand!

```python
>       return getattr(super().__getattr__("array"), attr)
E       AttributeError: type object 'super' has no attribute '__getattr__'
``